File: EmployeeAPI/EmployeeAPI/Services/EmployeeService.cs
public class EmployeeService
{
    private readonly List<Employee> _employees = new();
    private int _nextId = 1;

    public List<Employee> GetAll() => _employees;
    public Employee? GetById(int id) => _employees.FirstOrDefault(e => e.Id == id);
    public Employee Create(Employee emp)
    {
        emp.Id = _nextId++;
        _employees.Add(emp);
        return emp;
    }
    public bool Update(int id, Employee updated)
    {
        var emp = GetById(id);
        if (emp == null) return false;
        emp.FirstName = updated.FirstName;
        emp.LastName = updated.LastName;
        emp.Email = updated.Email;
        emp.Position = updated.Position;
        return true;
    }
    public bool Delete(int id)
    {
        var emp = GetById(id);
        if (emp == null) return false;
        _employees.Remove(emp);
        return true;
    }
}

File: employee-app/src/app/employee/services/employee.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Employee } from '../models/employee.model';
import { Observable } from 'rxjs';

@Injectable({ providedIn: 'root' })
export class EmployeeService {
  private readonly apiUrl = 'https://localhost:7235/api/employee';

  constructor(private http: HttpClient) {}

  getAll(): Observable<Employee[]> {
    return this.http.get<Employee[]>(this.apiUrl);
  }

  getById(id: number): Observable<Employee> {
    return this.http.get<Employee>(`${this.apiUrl}/${id}`);
  }

  create(emp: Employee): Observable<Employee> {
    return this.http.post<Employee>(this.apiUrl, emp);
  }

  update(id: number, emp: Employee): Observable<void> {
    return this.http.put<void>(`${this.apiUrl}/${id}`, emp);
  }

  delete(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}
